File: HackerEarth/collision.py
'''
ans = []
        for new in asteroids:
            while ans and new < 0 < ans[-1]:
                if ans[-1] < -new:
                    ans.pop()
                    continue
                elif ans[-1] == -new:
                    ans.pop()
                break
            else:
                ans.append(new)
        return ans
'''
for _ in range(int(input())):
    ast = [int(x) for x in input().split()]
    stack = []
    ans = []
    for this in ast:
        if this > 0:
            stack.append(this)
        else:
            killed = False
            while stack:
                if abs(this) < stack[-1]:
                    killed = True
                    break
                elif -this == stack[-1]:
                    killed = True
                    stack.pop()
                    break
                else:
                    stack.pop()
            if not killed:
                ans.append(this)
    ans.extend(stack)
    if len(ans)==0:
        print("NOTHING")
    else:
        print(*ans)

File: HackerEarth/collision.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var noc, noe int
	fmt.Scan(&noc)
	for i := 0; i < noc; i++ {
		fmt.Scan(&noe)
		ast := make([]int, noe)
		for j := 0; j < noe; j++ {
			fmt.Scan(&ast[j])
		}
		var ans []int
		for _, element := range ast {
			if len(ans) == 0 || element >= 0 || ans[len(ans)-1]*element >= 0 {
				ans = append(ans, element)
				continue
			}
			for (len(ans) > 0) && (ans[len(ans)-1] >= 0) && (ans[len(ans)-1]+element < 0) {
				ans = ans[:len(ans)-1]
			}

			if len(ans) == 0 || ans[len(ans)-1] < 0 {
				ans = append(ans, element)
				continue
			}
			if ans[len(ans)-1]+element == 0 {
				ans = ans[:len(ans)-1]
			}
		}

		if len(ans) == 0 {
			fmt.Println("NOTHING")
		} else {
			fmt.Println(strings.Trim(fmt.Sprint(ans), "[]"))
		}

	}

}
